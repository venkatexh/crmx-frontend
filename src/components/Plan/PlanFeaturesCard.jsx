import '../../sass/components/plan/planCard.component.scss';

const PlanFeaturesCard = ({plan}) => {
  // const {name, contacts, emails, featuresYes, featuresNo} = plan;
  const componentToRender = () => {
    return plan === {} ? <></>
      :
      <div className={'featuresCard'}>
        <div className={'planName'}>{plan.name}</div>
        <div className={'featuresContainer'}>
          <div className={'featuresLeft'}>
            <div className={'counts'}>{plan.contacts} contacts</div>
            <div className={'counts'}>{plan.emails} email sends per month</div>
          </div>
          <div className={'featuresRight'}>
            {plan?.featuresYes.map((feature, idx) => {
              return (
                <div key={idx} className={'feature'}>
                  <img src={'/icons/tick.png'} className={'icon'} alt={'icon'}/>
                  <div className={'featureText'}>{feature}</div>
                </div>
              )
            })}
            {plan?.featuresNo.map((feature, idx) => {
              return (
                <div key={idx} className={'feature'}>
                  <img src={'/icons/cross.png'} className={'icon'} alt={'icon'}/>
                  <div className={'featureText'}>{feature}</div>
                </div>
              )
            })}
          </div>
        </div>
      </div>
  }
  return (
    <>{componentToRender()}</>
  );
}

export default PlanFeaturesCard;