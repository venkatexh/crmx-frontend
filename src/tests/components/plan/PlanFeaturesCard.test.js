import { shallow } from "enzyme";
import React from "react";
import PlanFeaturesCard from "../../../components/Plan/PlanFeaturesCard";

it("Plan features card component snapshot", () => {
  expect(
    shallow(
      <PlanFeaturesCard
        plan={{
          name: "test plan",
          contacts: "500",
          emails: "10000",
          featuresYes: ["feature1yes", "feature2yes"],
          featuresNo: ["feature1no", "feature2no"],
        }}
      />
    )
  ).toMatchSnapshot();
});
