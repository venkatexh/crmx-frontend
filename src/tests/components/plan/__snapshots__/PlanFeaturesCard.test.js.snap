// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plan features card component snapshot 1`] = `ShallowWrapper {}`;
